src/engine/GameObjects/MapNode.cxx

```diff
@@ -201,54 +201,82 @@ bool MapNode::isPlacableOnSlope(const std::string &tileID) const
 bool MapNode::isPlacementAllowed(const std::string &newTileID) const
 {
   TileData *tileData = TileManager::instance().getTileData(newTileID);
-  const Layer layer = TileManager::instance().getTileLayer(newTileID);
 
   if (tileData)
   {
+    const Layer layer = TileManager::instance().getTileLayer(newTileID);
+    // layer specific checks:
     switch (layer)
     {
+    case Layer::ZONE:
+      // zones can overplace themselves and everything else
+      return true;
     case Layer::ROAD:
-      if (isLayerOccupied(Layer::ROAD))
-      {
-        return true;
-      }
-      else if ((isLayerOccupied(Layer::BUILDINGS) && (m_mapNodeData[Layer::BUILDINGS].tileData->category != "Flora")) ||
-               isLayerOccupied(Layer::WATER) || !isPlacableOnSlope(newTileID))
-      {
+      if ((isLayerOccupied(Layer::BUILDINGS) && (m_mapNodeData[Layer::BUILDINGS].tileData->category != "Flora")) ||
+          isLayerOccupied(Layer::WATER) || !isPlacableOnSlope(newTileID))
+      { // roads cannot be placed:
+        // - on buildings that are not category flora.
+        // - on water
+        // - on slopetiles that don't have a tileID
         return false;
       }
+      return true;
+    case Layer::GROUND_DECORATION:
+      if (m_mapNodeData[Layer::GROUND_DECORATION].tileData || m_mapNodeData[Layer::BUILDINGS].tileData)
+      { // allow placement of ground decoration on existing ground decoration and on buildings.
+        return true;
+      }
+      break;
     case Layer::BUILDINGS:
+      TileData *tileDataBuildings = m_mapNodeData[Layer::BUILDINGS].tileData;
+      if (tileDataBuildings && tileDataBuildings->isOverPlacable)
+      { // buildings with overplacable flag
+        return true;
+      }
       if (isLayerOccupied(Layer::ROAD))
-      {
+      { // buildings cannot be placed on water
```

Comment:
you are stating that buildings cannot be placed on water and you are checking ROAD layer
is this by mistake ?

Reply:
copy and paste error in the comment 😅  thanks